tool/src/dart/mod.rs

```diff
@@ -379,7 +415,15 @@ impl<'a, 'cx> TyGenContext<'a, 'cx> {
                 param_types_ffi_cast.push(self.formatter.fmt_usize(true).into());
                 param_names_ffi.push(format!("{param_name}Length").into());
 
-                let view_expr = self.gen_dart_to_c_for_type(&param.ty, param_name.clone());
+                let view_expr = self.gen_dart_to_c_for_type(&param.ty, param_name.clone(), None);
+
+                let is_borrowed = match param_borrow_kind {
+                    ParamBorrowInfo::TemporarySlice => false,
+                    ParamBorrowInfo::BorrowedSlice => true,
+                    _ => unreachable!(
+                        "Slices must produce slice ParamBorrowInfo, found {param_borrow_kind:?}"
```

Comment:
Not a big fan of this error case existing. We already have the type, why does `ParamBorrowInfo` include the type again?

Reply:
because the different types contain different kinds of borrowing info. I'm not a fan of this case either but it seemed like the cleanest way to get info across, no matter what we do there will be some such assert _somewhere_

Reply:
Then should we get the borrow info from the `hir::Slice`?

Reply:
No because this function computes it all

Reply:
Basically, either we have this error case, or we return ParamBorrowInfo as a more structlike thing with is_borrowed and struct_borrow_info fields, with an assertion that struct_borrow_info is none for non-structs